### DISCUSSION TITLE
ISSUE: $checked is ignored for radio button if value is zero

### DISCUSSION THREAD

**Discussion by sgilberg** [spatie/laravel-html#196] [Q&A]
I'm in the process of migrating a project from laravelcollective/html to spatie/laravel-html, and I've hit a snag with radio buttons. Specifically, we have a number of places where the radio button is to switch between a value of 1 and 0. These should default to be unset when the form is blank, but it is always defaulting to checking the 0 option, _no matter what value I put in the $checked boolean argument_. For example:

```
<fieldset class="form-group">
    <legend class="control-label">Status</legend>
    <div class="form-check">
        {{ html()->radio('public', null, 1)->id('public_public')->class('form-check-input') }}
        {{ html()->label('Public', 'public_public')->class('form-check-label') }}
    </div>
    <div class="form-check">
        {{ html()->radio('public', null, 0)->id('public_draft')->class('form-check-input') }}
        {{ html()->label('Draft', 'public_draft')->class('form-check-label') }}
    </div>
</fieldset>
```

Yields this result:
<img width="139" alt="Screenshot 2023-08-30 at 1 45 11 PM" src="https://github.com/spatie/laravel-html/assets/669465/ff4eab5b-2667-4bd8-93da-633a63889ecf">

Even doing `html()->radio('public', true, 1)` and `html()->radio('public', false, 0)` yields the same result. This is especially problematic when I want the default to be the first option. Best workaround I can come up with is to chain the ->checked() method to the end, but that gets tedious doing the long-form check for model and old values.

The problem seems to lie in the `$this->old($name) == $value` comparison in the [radio() method](https://github.com/spatie/laravel-html/blob/4c24827ca8c40e1e0595c64e8220ff466b2b7397/src/Html.php#L409-L422), in which 0 == null evaluates to true. I was going to attempt a PR for this, but it seems a bit sticky, especially given that, while it would be a bug fix, it's also potentially a breaking change for some implementations.

**Comment by kamleshwebtech** [comment]
If it ignores and considers as False then Use '2' (Non zero value) instead of 0. This solved my issue. Thanks.

**Reply by sgilberg** [reply]
Thanks, I would consider this another workaround, but not a solution. It would require another layer of translation from the model value because 2 is not a false value, and in this use case the radio is being used for a boolean.
